pmd-core/src/main/java/net/sourceforge/pmd/cpd/CPDConfiguration.java

```diff
@@ -156,12 +158,21 @@ public Language convert(String languageString) {
         }
     }
 
-    @Parameter(names = { "--encoding", "-e" }, description = "Character encoding to use when processing files", required = false)
+
+    /**
+     * @deprecated Use {@link #setSourceEncoding(String)} instead
+     */
+    @Deprecated
     public void setEncoding(String encoding) {
-        this.encoding = encoding;
         setSourceEncoding(encoding);
     }
 
+    @Parameter(names = { "--encoding", "-e" }, description = "Character encoding to use when processing files", required = false)
+    @Override
+    public void setSourceEncoding(String sourceEncoding) {
+        super.setSourceEncoding(sourceEncoding);
+    }
+
```

Comment:
what's the point of adding this override? This method should still be removed in PMD 7.

Reply:
This is to add the annotation `@Parameter` to this method...

Reply:
yeah, but why not keep it on the other (deprecated) method? We are going to remove both, the method and the annotation as part of PMD 7, as JCommander is no more.

Reply:
done